### PR TITLE
COTR-2 Added menu

### PR DESCRIPTION
**Pull request by disusen**

### CODE REVIEW

File: Scenes/Menus/Main/MainMenu.tscn [hunk 1]
```diff
@@ -0,0 +1,75 @@
+[gd_scene load_steps=2 format=3 uid="uid://pnp7uy840tdl"]
+
+[ext_resource type="Script" path="res://Scenes/Menus/Main/MainMenu.cs" id="1_buvme"]
+
+[node name="MainMenu" type="Node"]
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.tscn:5]: MainMenu scene node type should be "Control", not "Node"

File: Scenes/Menus/Main/MainMenu.tscn [hunk 2]
```diff
@@ -0,0 +1,75 @@
+[gd_scene load_steps=2 format=3 uid="uid://pnp7uy840tdl"]
+
+[ext_resource type="Script" path="res://Scenes/Menus/Main/MainMenu.cs" id="1_buvme"]
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.tscn:3]: Move MainMenu.cs and MainMenu.tscn files to "res://Scenes/UI/MainMenu/"

File: Scenes/Menus/Main/MainMenu.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+using Godot;
+using System;
+
+public partial class MainMenu : Godot.CanvasLayer
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.cs:4]: MainMenu node should Inherit from Control not "Godot.CanvasLayer"

File: Scenes/Menus/Main/MainMenu.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+{
+	// Called when the node enters the scene tree for the first time.
+	public override void _Ready()
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.cs:7]: Remove or comment out _Ready method

File: Scenes/Menus/Main/MainMenu.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	{
+	}
+
+	// Called every frame. 'delta' is the elapsed time since the previous frame.
+	public override void _Process(double delta)
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.cs:12]: Remove _Process method

File: Scenes/Menus/Main/MainMenu.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	{
+	}
+	
+	private void _on_button_start_new_pressed()
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.cs:16]: Use CamelCase for files, methods and variables (except private and protected variables should start with lower case letter). For example you should change "_on_button_start_new_pressed" to "ButtonStartNewPressed" after this you also need to reconnect button signals  with changed name to MainMenu.cs.

File: Scenes/Menus/Main/MainMenu.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	{
+		GetTree().ChangeSceneToFile("res://Scenes/Maps/Main/Main.tscn");
+	}
+	
+	private void _on_button_load_pressed()
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.cs:21]: CamelCase

File: Scenes/Menus/Main/MainMenu.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	{
+		// Replace with function body.
+	}
+
+	private void _on_button_settings_pressed()
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.cs:26]: CamelCase

File: Scenes/Menus/Main/MainMenu.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+	{
+		GetTree().ChangeSceneToFile("res://Scenes/Menus/Settings/Settings.tscn");
+	}
+
+	private void _on_button_quit_pressed()
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.cs:31]: CamelCase

File: Scenes/Menus/Settings/Settings.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+	}
+
+	// Called every frame. 'delta' is the elapsed time since the previous frame.
+	public override void _Process(double delta)
+	{
+	}
+	
+	private void _on_button_back_pressed()
+	{
+		GetTree().ChangeSceneToFile("res://Scenes/Menus/Main/MainMenu.tscn");
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/Settings.cs:18]: Change path after moving MainMenu scene

File: Scenes/Menus/Main/MainMenu.cs [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+	}
+	
+	private void _on_button_load_pressed()
+	{
+		// Replace with function body.
+	}
+
+	private void _on_button_settings_pressed()
+	{
+		GetTree().ChangeSceneToFile("res://Scenes/Menus/Settings/Settings.tscn");
```
Comment (Elijuszek) [inline Scenes/Menus/Main/MainMenu.cs:28]: Path should be "res://Scenes/UI/Menus/Settings/Settings.tscn"

File: Scenes/Menus/Settings/AudioContainer.cs [hunk 12]
```diff
@@ -0,0 +1,52 @@
+using Godot;
+using System;
+
+public partial class AudioContainer : VBoxContainer
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/AudioContainer.cs:4]: AudioContainer should Inherit from Control node

File: Scenes/Menus/Settings/AudioContainer.cs [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+	
+	// Called when the node enters the scene tree for the first time.
+	public override void _Ready()
+	{
+		musicBus = AudioServer.GetBusIndex("Music");
+		sfxBus = AudioServer.GetBusIndex("SFX");
+	}
+
+	// Called every frame. 'delta' is the elapsed time since the previous frame.
+	public override void _Process(double delta)
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/AudioContainer.cs:17]: Remove _Process method

File: Scenes/Menus/Settings/AudioContainer.cs [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+	{
+	}
+	
+	/// <summary>
+	/// Method for enabling or disabling music bus
+	/// </summary>
+	private void _on_music_toggle_toggled(bool button_pressed)
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/AudioContainer.cs:24]: CamelCase

File: Scenes/Menus/Settings/AudioContainer.cs [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+	{
+		AudioServer.SetBusMute(musicBus, !AudioServer.IsBusMute(musicBus));
+	}
+
+	/// <summary>
+	/// Method for enabling or disabling sfx bus
+	/// </summary>
+	private void _on_sfx_toggle_toggled(bool button_pressed)
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/AudioContainer.cs:32]: CamelCase

File: Scenes/Menus/Settings/AudioContainer.cs [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+	{
+		AudioServer.SetBusMute(sfxBus, !AudioServer.IsBusMute(sfxBus));
+	}
+	
+	/// <summary>
+	/// Music volume slider
+	/// </summary>
+	private void _on_music_slider_value_changed(double value)
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/AudioContainer.cs:40]: CamelCase

File: Scenes/Menus/Settings/AudioContainer.cs [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+	{
+		AudioServer.SetBusVolumeDb(musicBus, (float)value);
+	}
+	
+	/// <summary>
+	/// SFX volume slider
+	/// </summary>
+	private void _on_sfx_slider_value_changed(double value)
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/AudioContainer.cs:48]: CamelCase

File: Scenes/Menus/Settings/ResolutionContainer.cs [hunk 18]
```diff
@@ -0,0 +1,52 @@
+using Godot;
+using System;
+using System.Collections.Generic;
+
+public partial class ResolutionContainer : VBoxContainer
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/ResolutionContainer.cs:5]: ResolutionContainer Should Inherit from Control or CanvasLayer node

File: Scenes/Menus/Settings/ResolutionContainer.cs [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+	
+	// Called when the node enters the scene tree for the first time.
+	public override void _Ready()
+	{
+		resolutionMenu = GetNode<OptionButton>("ResolutionDropDown");
+		AddItems();
+	}
+
+	// Called every frame. 'delta' is the elapsed time since the previous frame.
+	public override void _Process(double delta)
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/ResolutionContainer.cs:19]: Remove empty _Process method

File: Scenes/Menus/Settings/ResolutionContainer.cs [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+		foreach (var resolution in Resolutions) 
+		{
+			resolutionMenu.AddItem(resolution.Key);
+		}
+	}
+	
+	/// <summary>
+	/// Window resolution drop down menu
+	/// </summary>
+	private void _on_resolution_drop_down_item_selected(long index)
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/ResolutionContainer.cs:47]: CamelCase

File: Scenes/Menus/Settings/Settings.cs [hunk 21]
```diff
@@ -0,0 +1,20 @@
+using Godot;
+using System;
+
+public partial class Settings : Node
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/Settings.cs:4]: Settings node should inherit from Control node

File: Scenes/Menus/Settings/Settings.cs [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+{
+	// Called when the node enters the scene tree for the first time.
+	public override void _Ready()
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/Settings.cs:7]: Remove empty _Ready method

File: Scenes/Menus/Settings/Settings.cs [hunk 23]
```diff
@@ -0,0 +1,20 @@
...
+	{
+	}
+
+	// Called every frame. 'delta' is the elapsed time since the previous frame.
+	public override void _Process(double delta)
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/Settings.cs:12]: Remove empty _Process method

File: Scenes/Menus/Settings/ResolutionContainer.cs [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+		{ "1366x768", new Vector2I(1366, 768) },
+		{ "1280x720", new Vector2I(1280, 720) },
+		{ "1024x768", new Vector2I(1024, 768) },
+		{ "800x600", new Vector2I(800, 600) }
+	};
+	
+	/// <summary>
+	/// Adds dictionary items to the resolution drop down menu
+	/// </summary>
+	public void AddItems()
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/ResolutionContainer.cs:36]: AddItems method should be private

File: Scenes/Menus/Settings/Settings.cs [hunk 25]
```diff
@@ -0,0 +1,20 @@
...
+	{
+	}
+	
+	private void _on_button_back_pressed()
```
Comment (Elijuszek) [inline Scenes/Menus/Settings/Settings.cs:16]: CamelCase

File: Scenes/Menus/Settings/Settings.cs (hunk 10 above)
Comment (Elijuszek) [inline Scenes/Menus/Settings/Settings.cs:18]: Settings scene should be Added as a child on top of MainMenu scene. bcause ChangingScene is more performance expensive than Adding child node, also player won't  be able to open and change settings while he is in other scenes.

**Review by Elijuszek — CHANGES_REQUESTED** [review]
Need further enhancments before merging. Also methods should be fully commented using summary

File: Scenes/UI/Menus/Settings/Settings.cs [hunk 26]
```diff
@@ -0,0 +1,110 @@
+using Godot;
+using System;
+using System.Collections.Generic;
+
+public partial class Settings : Control
+{	
+	// audio variables
+	int musicBus;
```
Comment (Karolinskis) [inline Scenes/UI/Menus/Settings/Settings.cs:8]: Add access modifiers for better readability

**Review by Karolinskis — CHANGES_REQUESTED** [review]

**Review by Elijuszek — APPROVED** [review]

**Review by Elijuszek — APPROVED** [review]
LGTM!

**Review by Karolinskis — APPROVED** [review]

**Pull request merged**